### PR TITLE
fixes #5660 - use with_taxonomy_scope for OS media selection

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Comment by ohadlevy** [comment]
@isratrade can you review please?

**Comment by isratrade** [comment]
@stbenjam, what's the background behind this requested change. It goes against the guidelines that we decided for select boxes in forms.

**Comment by stbenjam** [comment]
@isratrade It's a similar issue as http://projects.theforeman.org/issues/5106. When configuring an OS (for a host), it shows _all_ the installation media because with_taxonomy_scope_override calls unscoped.  That's intended behavior?

File: app/helpers/hosts_and_hostgroups_helper.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def arch_oss
 
   def os_media
     return [] if @operatingsystem.blank?
-    @operatingsystem.media.with_taxonomy_scope_override(@location,@organization)
+    @operatingsystem.media.with_taxonomy_scope(@location,@organization)
```
Comment (isratrade) [inline app/helpers/hosts_and_hostgroups_helper.rb]: I suggest to add `:path_ids` as the 3rd argument similar to what I did here 
https://github.com/theforeman/foreman/pull/1435/files

Comment (stbenjam) [inline app/helpers/hosts_and_hostgroups_helper.rb]: Done

**Comment by mmoll** [comment]
tested, works for me :+1:

**Comment by isratrade** [comment]
ack from me

**Comment by isratrade** [comment]
merged as f15d04122eed5be747a16d4749ea00f5c38096e8. thanks @stbenjam !

**Pull request closed without merging**